### PR TITLE
Split linear sieve from arithmetic_functoin.md

### PR DESCRIPTION
**Pull request by PolarisChiba**

### CODE REVIEW

**Comment by harry900831** [comment]
Please resolve the above comments first.

**Comment by harry900831** [comment]
@PolarisChiba You hasn't resolved the comment.

**Review by harry900831 — CHANGES_REQUESTED** [review]

File: src/math/linear_sieve.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    if (i * p > n)
+      break;
+    primeFactor[i * p] = p;
+    if (i % p == 0)
+      break;
+  }
+}
+```
+
+## 數論函數篩法
```
[the remark this answers is not in the document]
Comment (PolarisChiba) [reply src/math/linear_sieve.md:81]: I tend to put it under arithmetic function now and maybe move it under "Seive" chapter if there is one in the future

File: src/math/linear_sieve.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
+# 線性篩
```
[the remark this answers is not in the document]
Comment (PolarisChiba) [reply src/math/linear_sieve.md:1]: I plan to do it in the future.

**Comment by harry900831** [comment]
Add linear sieve to SUMMAYR
-> SUMMARY

File: src/math/linear_sieve.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    primes.push_back(i);
+  for (auto p : primes) {
+    if (i * p > n)
+      break;
+    notPrime[i * p] = true;
+    if (i % p == 0)
+      break;
+  }
+}
+```
```
Comment (harry900831) [inline src/math/linear_sieve.md:23]: Make this a function.
Declare notPrime and primes as a vector

File: src/math/linear_sieve.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+對於任何合數 \\( x \\)，假設其質因數分解為 \\( p_1^{\alpha_1}p_2^{\alpha_2}\cdots p_m^{\alpha_m} \\)，其中 \\( p_1 \\) 是最小質因數。
+
+當 \\( i = \frac{x}{p_1} \\) 時，由於 \\( p_1 < i \\)，因此在進入 ``for (auto : primes)`` 之前 \\( p_1 \\) 就會被放到 ``primes`` 裡。
+
+所有小於 \\( p_1 \\) 的質數都無法整除 \\( i \\)，因此會順利執行到 ``notPrime[i * p] = true`` 這行而將 \\( x \\) 設成合數。
+
+> 所有合數 \\( x \\) 都只會被篩到一次。
+
+對於並非最小的質數 \\( p_2 \\)，當 \\( i = \frac{x}{p_2} \\) 的時候，因為 \\( i\mod p_2 = 0 \\)，因此會提早 ``break``。
```
Comment (harry900831) [inline src/math/linear_sieve.md:47]: use /bmod instead of /mod

File: src/math/linear_sieve.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+
+執行完以上程式碼後，``notPrime`` 為 ``true`` 的話代表是合數，為 ``false`` 代表是質數。
+
+而 ``primes`` 裡則會是 \\( 1\sim n \\) 的質數由小到大的排列。
+
+接下來我們會證明其正確性與時間複雜度。
+
+證明：
+
+> 所有質數的 ``notPrime`` 值都會是 ``false``。
```
Comment (harry900831) [inline src/math/linear_sieve.md:33]: 改成以下可能比較通順：
對於所有質數 i， 經過上述計算後，notPrime[i] 等於 false。

File: src/math/linear_sieve.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+
+注意到 ``notPrime[i * p]`` 中的 ``i`` 和 ``p`` 都大於 \\( 1 \\)，因此只有合數的 ``notPrime`` 值會被設為 ``true``。
+
+> 所有合數的 ``notPrime`` 值都會是 ``true``。
```
Comment (harry900831) [inline src/math/linear_sieve.md:37]: Ditto

File: src/math/linear_sieve.md (hunk 4 above)
Comment (harry900831) [inline src/math/linear_sieve.md:47]: 這證明怪怪的吧？

File: src/math/linear_sieve.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+  }
+  for (auto p : primes) {
+    if (i * p > n)
+      break;
+    primeFactor[i * p] = p;
+    if (i % p == 0)
+      break;
+  }
+}
+```
```
Comment (harry900831) [inline src/math/linear_sieve.md:79]: Ditto, make it a function.

File: src/math/linear_sieve.md [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+
+## 數論函數篩法
+
+許多的數論函數都能夠透過線性篩進行打表。
+
+一般而言，給定函數 \\( f \\)要求 \\( f(n) \\)，假設 \\( p \\) 是 \\( n \\) 的最小質因數。
```
Comment (harry900831) [inline src/math/linear_sieve.md:85]: need a space after \\( f \\)

File: src/math/linear_sieve.md [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+
+```cpp
+mu[1] = 1;
+for (int i = 2; i <= n; ++i) {
+  if (i % (1LL * primeFactor[i] * primeFactor[i]) == 0)
+    mu[i] = 0;
+  else
+    mu[i] = -mu[i / primeFactor[i]];
+}
+```
```
Comment (harry900831) [inline src/math/linear_sieve.md:127]: Ditto

File: src/math/linear_sieve.md [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+
+```cpp
+phi[1] = 1;
+for (int i = 2; i <= n; ++i) {
+  if (i % (1LL * primeFactor[i] * primeFactor[i]) == 0)
+    phi[i] = phi[i / primeFactor[i]] * primeFactor[i];
+  else
+    phi[i] = phi[i / primeFactor[i]] * (primeFactor[i] - 1);
+}
+```
```
Comment (harry900831) [inline src/math/linear_sieve.md:169]: Ditto

**Review by harry900831 — CHANGES_REQUESTED** [review]

File: src/math/linear_sieve.md (hunk 4 above)
Comment (harry900831) [reply src/math/linear_sieve.md:47]: https://math.meta.stackexchange.com/questions/20988/modulo-operator-rendering-poorly

File: src/math/linear_sieve.md [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+而在此章節，我們將會介紹 \\( \mathcal{O}(n) \\) 的演算法，也就是線性篩。
+
+## 質數篩法
+
+直接看 code。
+
+```cpp
+vector<int> LinearPrimeSieve(int n) {
+  vector<int> primes;
+  vector<int> notPrime(n + 1, false);
```
Comment (harry900831) [inline src/math/linear_sieve.md:14]: vector\<bool\>

File: src/math/linear_sieve.md (hunk 6 above)
Comment (harry900831) [reply src/math/linear_sieve.md:43]: You didn't change this

File: src/math/linear_sieve.md [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+
+我們會分成三個情況討論：
+
+\begin{cases}
+n=1\\\\
+p^2\mid n\\\\
+p\mid n
+\end{cases}
+
+特別注意到第三個情況也可以視作第一、二情況的 Otherwise。
```
Comment (harry900831) [inline src/math/linear_sieve.md:106]: Need more explanation

File: src/math/linear_sieve.md [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+    if (i % (1LL * primeFactor[i] * primeFactor[i]) == 0)
+      phi[i] = phi[i / primeFactor[i]] * primeFactor[i];
+    else
+      phi[i] = phi[i / primeFactor[i]] * (primeFactor[i] - 1);
+  }
+  return phi;
+}
+```
+
+## 附註
```
Comment (harry900831) [inline src/math/linear_sieve.md:190]: Summary

**Review by harry900831 — CHANGES_REQUESTED** [review]

**Review by harry900831 — APPROVED** [review]

**Pull request merged**